Top level gem azure_sdk bundling azure management and data plane

vishrutshah opened the pull request:
Top level gem azure_sdk bundling azure management and data plane.

azurecla commented:
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

stuartpreston commented:
Wow this looks like an interesting commit...! - does this mean the individual resource gems will not be loaded to RubyGems in future? /cc @devigned

devigned commented:
@stuartpreston I believe that the gems will still be released as they are, but we'll also introduce a roll up gem to bring them together and provide a simpler interface when using multiple clients.

@vishrutshah or @veronicagg please correct me if I've misspoke.

devigned commented on azure_sdk/lib/azure_sdk/client.rb:
```diff
@@ -0,0 +1,537 @@
+    # @param options [Array] filters to be applied to the HTTP requests.
+    # @return [Azure::ARM::Authorization::AuthorizationManagementClient] Azure ARM Authorization Management Client.
+    #
+    def authorization(base_url = nil, options = nil)
```
a little bit of meta programing would go a long way here :)

vishrutshah replied:
@devigned That's the killer idea. Ruby Meta programming rocks :) 

It'd appreciate if you can spare sometime to review this PR. Thanks!

vishrutshah commented:
@veronicagg Feel free to review this PR as you get a chance. Thanks!

devigned commented:
LGTM 👍

veronicagg commented on azure_sdk/lib/azure_sdk/client.rb:
```diff
@@ -0,0 +1,165 @@
+    end
+
+    #
+    # Check if this client is configured with the some options.
```
should this description be updated?

vishrutshah replied:
Yup. Corrected.

---
In reply to: [76343363](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76343363) [](ancestors = 76343363)

veronicagg commented on azure_sdk/lib/azure_sdk/version.rb:
```diff
@@ -0,0 +1,7 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+module Azure
+    VERSION = '0.5.0' # File.read(File.expand_path('../../../../ARM_VERSION', __FILE__)).strip
```
should the comment be removed?

vishrutshah replied:
Yup not required.

---
In reply to: [76344267](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344267) [](ancestors = 76344267)

veronicagg commented on azure_sdk/spec/azure_sdk_spec.rb:
```diff
@@ -0,0 +1,125 @@
+    end
+
+    it 'should initialize by defaulting to ENV' do
+      arm_client = Azure::ARM.client({})
```
do we need ({})?

vishrutshah replied:
nope. removed

---
In reply to: [76344399](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344399) [](ancestors = 76344399)

veronicagg commented on azure_sdk/spec/azure_sdk_spec.rb:
```diff
@@ -0,0 +1,125 @@
+          @tenant_id, @client_id, @client_secret, @active_directory_settings))
+      Azure::ARM.subscription_id = @subscription_id
+
+      verify_management_clients(Azure::ARM)
```
should we be validating that credentials of at least one client matches what you set it to? and would that be enough validation for this test?

vishrutshah replied:
Sure

---
In reply to: [76344795](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344795) [](ancestors = 76344795)

veronicagg commented:
@vishrutshah Left minor comments, rest looks great to me, thanks!

vishrutshah commented:
@veronicagg Thanks for the review! I've addressed the comments. Feel free to skim through it again. 

---
In reply to: [242583285](https://github.com/Azure/azure-sdk-for-ruby/pull/433#issuecomment-242583285) [](ancestors = 242583285)
